test/parallel/test-worker-prof.js

```diff
@@ -1,40 +1,75 @@
 'use strict';
-require('../common');
+const common = require('../common');
 const tmpdir = require('../common/tmpdir');
 const fs = require('fs');
 const assert = require('assert');
 const { join } = require('path');
 const { spawnSync } = require('child_process');
-const { Worker } = require('worker_threads');
 
 // Test that --prof also tracks Worker threads.
 // Refs: https://github.com/nodejs/node/issues/24016
 
 if (process.argv[2] === 'child') {
-  const spin = `
-  const start = Date.now();
-  while (Date.now() - start < 1000);
+  let files = fs.readdirSync(tmpdir.path);
+  const plog = files.filter((name) => /\.log$/.test(name))[0];
+  if (plog === undefined) {
+    console.error('`--prof` did not produce a profile log for parent thread!');
+    process.exit(1);
+  }
+  const pingpong = `
+  let counter = 0;
+  const { Worker, parentPort  } = require('worker_threads');
+  parentPort.on('message', (m) => {
+    if (counter++ === 1024)
+      process.exit(0);
+     parentPort.postMessage(
+       m.toString().split('').reverse().toString().replace(/,/g, ''));
+  });
   `;
-  new Worker(spin, { eval: true });
-  eval(spin);
-  return;
-}
 
-tmpdir.refresh();
-spawnSync(process.execPath, ['--prof', __filename, 'child'],
-          { cwd: tmpdir.path });
-const files = fs.readdirSync(tmpdir.path);
-const logfiles = files.filter((name) => /\.log$/.test(name));
-assert.strictEqual(logfiles.length, 2);  // Parent thread + child thread.
+  const { Worker } = require('worker_threads');
+  const data = 'x'.repeat(1024);
+  const w = new Worker(pingpong, { eval: true });
+  w.on('message', (m) => {
+    w.postMessage(m.toString().split('').reverse().toString().replace(/,/g, ''));
+  });
+
+  w.on('exit', common.mustCall(() => {
+    files = fs.readdirSync(tmpdir.path);
+    const wlog = files.filter((name) => /\.log$/.test(name) && name !== plog)[0];
+    if (wlog === undefined) {
+      console.error('`--prof` did not produce a profile log' +
+                    ' for worker thread!');
+      process.exit(1);
+    }
+    process.exit(0);
+  }));
+  w.postMessage(data);
+} else {
+  tmpdir.refresh();
+  const spawnResult = spawnSync(
+    process.execPath, ['--prof', __filename, 'child'],
+    { cwd: tmpdir.path });
+  assert.strictEqual(spawnResult.stderr.toString(), '',
+                     `child exited with an error: ${spawnResult}`);
+  assert.strictEqual(spawnResult.signal, null,
+                     `child exited with signal: ${spawnResult}`);
```

Comment:
`${spawnResult}` will end up displaying `[object Object]` unfortunately. You might have to do something like `${util.inspect(spawnResult)}` if you want something useful.

Reply:
thanks. Is there any way to explode the internal arrays as well? with this change I get the primitive fields, but not the Buffers:
```js
assert.strictEqual(spawnResult.status, 0,
                     `child exited with non-zero status: ${util.inspect(spawnResult, {depth: 3})}`);
```
```trace
{ status: 1,
  signal: null,
  output: [ null, <Buffer >, <Buffer > ],
  pid: 72177,
  stdout: <Buffer >,
  stderr: <Buffer > 
}
```

Reply:
How about passing `encoding: 'utf8'` option to `spawnSync` so that `stdout` and `stdin` are String objects instead of Buffer objects?

Reply:
@richardlau - addressed your review comments, ptal.